libcontainer/cgroups/fs/cpuset.go

```diff
@@ -14,6 +16,41 @@ import (
 	"github.com/opencontainers/runc/libcontainer/configs"
 )
 
+var (
+	cpusetLock   sync.Mutex
+	cpusetPrefix string
+	cpusetInit   bool
+)
+
+func getCpusetPrefix() string {
+	cpusetLock.Lock()
+	defer cpusetLock.Unlock()
+
+	if cpusetInit {
+		return cpusetPrefix
+	}
+
+	cpusetInit = true
+	cpusetPrefix = "cpuset."
+
+	file, err := os.Open("/proc/self/mountinfo")
+	if err != nil {
+		return cpusetPrefix
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, "cpuset") && strings.Contains(line, "cgroup") && strings.Contains(line, "noprefix") {
+			cpusetPrefix = ""
+			break
+		}
+	}
```

Comment:
Please find a way to do this without parsing `/proc/self/mountinfo`, as this is expensive and error-prone (on older kernels -- and we're using cgroup v1 on older kernels).

Reply:
I think a less invasive approach would be to make a `cpuset`-specific wrapper of `fscommon.GetCgroupParamUint`/`cgroups.WriteFile` that tries `cpuset.foo` first and then `foo` if there was an `ENOENT`. Yeah, it would double the number of syscalls needed but it would have no overhead or other downsides for regular systems.

Also the logic here is wrong (a `tmpfs` mount on a directory called `cpuset-cgroup-noprefix` will cause this code to think that there is a `cgroup` mount with those settings). As @kolyshkin said, getting this right is quite complicated and even in the best case where you get the parsing correct, there are cases where you can't get a proper snapshot of `/proc/self/mountinfo`. `statmount(2)` will give you the right answer but that's only available on very new kernels (there isn't even a `golang.org/x/sys/unix` wrapper of it yet).

Reply:
Done. I touch cpuset.foo then foo if it fails. If any successed it is stored as default.